utils/vqm2blif/src/base/cleanup.cpp

```diff
@@ -848,8 +836,64 @@ void remove_node ( t_node* node, t_node** nodes, int original_num_nodes ) {
 	}
 	
 	VTR_ASSERT(found);
-	oneluts_elim++;
 }
 
 //============================================================================================
 //============================================================================================
+
+void reorganize_module_node_list(int original_number_of_nodes, int number_of_nodes_eliminated, t_node** module_node_list, t_module* module)
+{
+	/*
+        The purpose of this function is to regorganize the nodes array by filling in gaps with the last available elements in the array to save CPU time.
+
+		The node array provided to this function is expected to have elements deleted within. THis essentially creates gaps in the array. So we fill in the gaps of this array and ensure it is a continuous list of nodes.
```

Comment:
explain that the node array is module_node_list, and that deleted elements within it are NULL.
Do you need original_number_of_nodes, or would that be equal to module->number_of_nodes?
number_of_nodes_eliminated seems dangerous to pass in.  It should be redundant with the number of NULL entries in the module_node_list, so it doesn't have to be passed.  It isn't asserted to match the number of NULLs eliminated, so it is also a bit dangerous -- if it doesn't match then the module_node_list could be corrupted.

Reply:
This has been updated now. number_of_nodes and nodes have been removed. Additionally, the number_of_nodes_eliminated has also been removed as a parameter and we are now determining this within the function.

Reply:
the function "remove_one_lut_nodes" also had unnecessary parameters that could have been found through "module", so the function was also changed as well.